internal/ted448/point.go

```diff
@@ -0,0 +1,179 @@
+package ted448
+
+import (
+	"crypto/subtle"
+	"fmt"
+
+	fp "github.com/cloudflare/circl/math/fp448"
+)
+
+// Point defines a point on the ted448 curve using extended projective
+// coordinates. Thus, for any affine point (x,y) it holds x=X/Z, y = Y/Z, and
+// T = Ta*Tb = X*Y/Z.
+type Point struct{ X, Y, Z, Ta, Tb fp.Elt }
+
+type prePointAffine struct{ addYX, subYX, dt2 fp.Elt }
+
+type prePointProy struct {
+	prePointAffine
+	z2 fp.Elt
+}
+
+func (P Point) String() string {
+	return fmt.Sprintf("x: %v\ny: %v\nta: %v\ntb: %v\nz: %v", P.X, P.Y, P.Ta, P.Tb, P.Z)
+}
+
+// cneg conditionally negates the point if b=1.
+func (P *Point) cneg(b uint) {
+	t := &fp.Elt{}
+	fp.Neg(t, &P.X)
+	fp.Cmov(&P.X, t, b)
+	fp.Neg(t, &P.Ta)
+	fp.Cmov(&P.Ta, t, b)
+}
+
+// Double updates P with 2P.
+func (P *Point) Double() {
+	// This is formula (7) from "Twisted Edwards Curves Revisited" by
+	// Hisil H., Wong K.KH., Carter G., Dawson E. (2008)
+	// https://doi.org/10.1007/978-3-540-89255-7_20
+	Px, Py, Pz, Pta, Ptb := &P.X, &P.Y, &P.Z, &P.Ta, &P.Tb
+	a, b, c, e, f, g, h := Px, Py, Pz, Pta, Px, Py, Ptb
```

Comment:
Note that you're overwriting P.Ta and P.Tb, but that that doesn't matter as it's not used in doubling.